### DISCUSSION TITLE
Get CRC Hash from S2 Header

### DISCUSSION THREAD

**Discussion by darthShadow** [klauspost/compress#782] [Q&A] [answered]
Hi,

Would it be possible to add a method to extract the CRC hash from the S2 header of an S2 compressed block?

Thanks.

**Comment by klauspost** [comment, marked as answer]
For S2 (and Snappy) a block doesn't have any checksums or content validation. 

On a Stream, each block is written with a checksum, that is a CRC32C, that is shifted: https://github.com/klauspost/compress/blob/master/s2/s2.go#L112-L117

This means that checksums here only represents each block, and you cannot get the total stream CRC.

If you just want *a* checksum and not a specific one, you could read read the stream and combine the checksums.

The framing format is the same as [Snappy framing format](https://github.com/google/snappy/blob/master/framing_format.txt), except S2 has a different [stream identifier](https://github.com/klauspost/compress/blob/master/s2/s2.go#L79).

**Reply by darthShadow** [reply]
That helps, thanks for the explanation.
